src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -384,7 +386,26 @@ public static IFullNodeBuilder UseBaseFeature(this IFullNodeBuilder fullNodeBuil
                     services.AddSingleton<ISelfEndpointTracker, SelfEndpointTracker>();
 
                     // Consensus
-                    services.AddSingleton<IConsensusManager, ConsensusManager>();
+                    services.AddSingleton<IConsensusManager>(provider => new ConsensusManager(
```

Comment:
please add a comment why CM is created differently comparing to other components